src/analytic/GRD2D/GRD2DReconstruction.cxx

```diff
@@ -0,0 +1,518 @@
+/*
+    Copyright (C) 2025 University College London
+
+    This file is part of STIR.
+
+    SPDX-License-Identifier: Apache-2.0 AND License-ref-PARAPET-license
+
+    See STIR/LICENSE.txt for details
+*/
+/*!
+  \file
+  \ingroup GRD2D
+  \brief Implementation of class stir::GRD2DReconstruction
+
+  \author Dimitra Kyriakopoulou
+  \author Kris Thielemans
+
+*/
+  
+#include "stir/analytic/GRD2D/GRD2DReconstruction.h"
+#include "stir/VoxelsOnCartesianGrid.h"
+//#include "stir/RelatedViewgrams.h"
+//#include "stir/recon_buildblock/BackProjectorByBinUsingInterpolation.h"
+#include "stir/ProjDataInfoCylindrical.h"
+
+//kept for future work, i.e. for the commented out SSRB and arc-correction
+//#include "stir/ProjDataInfoCylindricalArcCorr.h"
+//#include "stir/ArcCorrection.h"
+//#include "stir/SSRB.h"
+//#include "stir/ProjDataInMemory.h"
+
+#include "stir/Bin.h"
+//#include "stir/round.h"
+#include "stir/display.h"
+//#include <algorithm>
+//#include "stir/IO/interfile.h"
+
+#include "stir/Sinogram.h" 
+//#include "stir/Viewgram.h"
+#include <complex> 
+//#include <math.h>
+#include "stir/numerics/fourier.h"
+#include "stir/numerics/fftshift.h"
+#include <boost/math/special_functions/bessel.hpp> 
+#include "stir/info.h"
+#include "stir/format.h"
+#include "stir/error.h"
+#include "stir/warning.h"
+
+//kept for future work
+//#ifdef STIR_OPENMP
+//#include <omp.h>
+//#endif
+//#include "stir/num_threads.h"
+
+#include <vector>
```

Comment:
need `<memory>` for `std::dynamic_pointer_cast`. (I guess it's already included, but best to be sure)